bundler/lib/bundler/definition.rb

```diff
@@ -587,7 +587,13 @@ def filter_specs(specs, deps)
     end
 
     def materialize(dependencies)
-      specs = resolve.materialize(dependencies)
+      specs = begin
+        resolve.materialize(dependencies)
+      rescue IncorrectLockfileDependencies => e
+        reresolve_without([e.spec])
```

Comment:
is it possible this leads to an infinite loop?

Reply:
I thought about it and considered adding infinite loop detection here, but I decided to skip it because I think it should never lead to an infinite loop, since we will be resolving with less and less locked specs until there are none, and the error can no longer be raised.

I'm not 100% sure though, so I could also add it just in case, and print the bug report template if it happens, and eventually remove the detection if no issues are reported.

Reply:
I added a basic infinite loop detection at 0927ae400b945e14514bee09b395740f47bb2513 that we can remove after a few releases if no problems are found 👍.